Sources/Errors.php

```diff
@@ -123,6 +123,19 @@ function log_error($error_message, $error_type = 'general', $file = null, $line
 		$smcFunc['db_error_insert']($error_info);
 		$last_error = $error_info;
 
+		// Get an error count, if necessary
+		if (!isset($context['num_errors']))
+		{
+			$query = $smcFunc['db_query']('', '
+				SELECT COUNT(id_error)
+				FROM {db_prefix}log_errors',
+				array()
+			);
+
+			list($context['num_errors']) = $smcFunc['db_fetch_row']($query);
+			$smcFunc['db_free_result']($query);
+		}
+
```

Comment:
Should have an `else` here.

Reply:
Nope it get last number from db,
but this part is called when a new error come in.
--> get the db number and add one to get the new count.

Reply:
The latest error was already inserted into the database on line 123 above, so the query retrieves the correct error number at that moment. Incrementing the variable after the query sets it to one more than the number of errors.

For example, suppose that the error log already has 3 errors recorded in it when the script starts. Then a new error occurs sometime before we reach `setupMenuContext()`. Here's what will happen:

1. `$smcFunc['db_error_insert']($error_info);` inserts the new error into the database. There are now 4 errors recorded in the database.
2. `list($context['num_errors']) = $smcFunc['db_fetch_row']($query);` sets `$context['num_errors']` to 4.
3. `$context['num_errors']++;` sets `$context['num_errors']` to 5.